homework/index.js

```diff
@@ -27,21 +27,159 @@
+      const noDescription = createAndAppend('li', resetInfoDiv, {});
+      createAndAppend('span', noDescription, {
+        text: 'Description: there is no description',
+        class: '',
+      });
+    } else {
+      const liDescription = createAndAppend('li', resetInfoDiv, {});
+      createAndAppend('span', liDescription, { text: 'Description: ' });
+      createAndAppend('span', liDescription, { text: getDescription, class: 'result' });
+    }
+
+    const link = createAndAppend('li', resetInfoDiv, {});
+    createAndAppend('span', link, { text: 'link: ' });
+    createAndAppend('a', link, {
+      text: dataName,
+      target: '_blank',
+      href: `https://github.com/HackYourFuture/${dataName}`,
+      class: 'result',
+    });
+    const liOfForks = createAndAppend('li', resetInfoDiv, {});
+    createAndAppend('span', liOfForks, { text: 'Forks: ' });
+    createAndAppend('span', liOfForks, { text: getForks, class: 'result' });
+    const liUpdateAt = createAndAppend('li', resetInfoDiv, {});
+    createAndAppend('span', liUpdateAt, { text: 'updated_at: ' });
+    createAndAppend('span', liUpdateAt, { text: getUpdatedAt, class: 'result' });
+
+    // call all contributors for this selected repository
+    fetchContributorsUrl(getContributorsUrl); // comment this line to stop getting contributors
+  }
+
+  function reloadAllRepositories(parentElement, data) {
+    const pre = document.getElementById('pre');
+    const parent = createAndAppend('div', parentElement, { class: 'bodyInformation' });
+    createAndAppend('div', parent, { id: 'info', class: 'info' });
+    createAndAppend('div', parent, { id: 'cont', class: 'cont' });
+
+    const selectRepository = createAndAppend('select', pre, { id: 'select', class: 'select' });
+
+    for (let i = 0; i < data.length; i++) {
+      createAndAppend('option', selectRepository, { text: data[i].name, value: i });
+    }
+    // sort list of repository's
+    sortList();
+
+    // reload default repository
+    reloadSelectedRepository(0, data);
+
+    // reload selected repository
+    selectRepository.onchange = function reloadOnceSelect() {
+      const indexOfSelectedRepo = this.value;
+      reloadSelectedRepository(indexOfSelectedRepo, data); // comment this line to stop getting information once change select list
+    };
+  }
+
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
+      root.className = 'root';
+      const pre = createAndAppend('pre', root, { class: 'pre', id: 'pre' });
+      createAndAppend('span', pre, { text: 'HYF Repositories', class: 'logoName' });
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        // we can use below code to sort list of repository instead of sort function above
+        // data.sort((x, y) => x.name.localeCompare(y.name));
```

Comment:
Just saw this comment after commenting on the same idea above. Yes, better to sort before rendering.